### PR TITLE
Properly convert html to text

### LINKED ISSUE #842: Some messages fail to load because of reply text computation
$(message.body).text() is falling apart in case the html is not proper

````
Uncaught Error: JQMIGRATE: Invalid selector string (XSS)
    at Error (native)
    at Function.x.extend.error (http://localhost:8080/core/vendor/jquery/jquery.min.js?v=bdeb9472ab60782eaecfda1168dd38da:4:4112)
    at e.fn.init (http://localhost:8080/core/vendor/jquery-migrate/jquery-migrate.min.js?v=bdeb9472ab60782eaecfda1168dd38da:2:2716)
    at x (http://localhost:8080/core/vendor/jquery/jquery.min.js?v=bdeb9472ab60782eaecfda1168dd38da:4:225)
    at eval (eval at evaluate (unknown source), <anonymous>:1:10)
    at Object.InjectedScript._evaluateOn (<anonymous>:895:55)
    at Object.InjectedScript._evaluateAndWrap (<anonymous>:828:34)
    at Object.InjectedScript.evaluateOnCallFrame (<anonymous>:954:21)
    at Mail.UI.loadMessage.loadMessageSuccess (http://localhost:8080/apps2/mail/js/mail.js?v=bdeb9472ab60782eaecfda1168dd38da:948:7)
    at Object.Mail.UI.loadMessage.Mail.Communication.fetchMessage.onSuccess (http://localhost:8080/apps2/mail/js/mail.js?v=bdeb9472ab60782eaecfda1168dd38da:1046:8)

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes #842

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
@jancborchardt @ChristophWurst @irgendwie please review - thx

File: js/mail.js [hunk 1]
```diff
@@ -872,6 +872,12 @@ var Mail = {
 			Mail.UI.setMessageActive(null);
 		};
 
+		this.htmlToText = function (html) {
+			var tmp = document.createElement("DIV");
```
Comment (ChristophWurst) [inline js/mail.js]: please use ``'`` quotes instead of ``"`` :-)

File: js/mail.js [hunk 2]
```diff
@@ -872,6 +872,12 @@ var Mail = {
...
+			tmp.innerHTML = html;
+			return tmp.textContent || tmp.innerText || "";
```
Comment (ChristophWurst) [inline js/mail.js]: same here

**Comment by ChristophWurst** [comment]
How can I reproduce the issue?

**Comment by DeepDiver1975** [comment]
> How can I reproduce the issue?

I had a bunch of html mails - let me see what I can do ....

**Comment by irgendwie** [comment]
LGTM :+1: still, there are some emails now, which are 'onelined', because they use `<br />` tags, which seem to be not handled with a newline.

File: js/mail.js (hunk 1 above)
Comment (jancborchardt) [inline js/mail.js]: Also, lowercase `div`?

**Comment by jancborchardt** [comment]
Fixes the issue for the email which I could not open. :+1:

**Comment by jancborchardt** [comment]
Huh? Travis just sent me an email that the build was broken, but it was green here?

**Comment by DeepDiver1975** [comment]
> Huh? Travis just sent me an email that the build was broken, but it was green here?

the errors are on master - related to gmail feedback

**Comment by jancborchardt** [comment]
Ah, k. In Travis it’s red for ownCloud 7 https://travis-ci.org/owncloud/mail/builds/69878112

**Pull request merged**